src/pydata_sphinx_theme/__init__.py

```diff
@@ -49,33 +50,61 @@ def update_config(app, env):
         )
 
     # Add an analytics ID to the site if provided
-    # Currently only supports the two types of Google Analytics id.
+
+    # deprecated options for Google Analytics
     gid = theme_options.get("google_analytics_id")
     if gid:
-        # In this case it is "new-style" google analytics
-        if "G-" in gid:
-            gid_js_path = f"https://www.googletagmanager.com/gtag/js?id={gid}"
-            gid_script = f"""
-                window.dataLayer = window.dataLayer || [];
-                function gtag(){{ dataLayer.push(arguments); }}
-                gtag('js', new Date());
-                gtag('config', '{gid}');
-            """
-        # In this case it is "old-style" google analytics
+        msg = (
+            "'google_analytics_id' is deprecated and will be removed in "
+            "version 0.11, please refer to the documentation "
+            "and use 'analytics' instead."
+        )
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+        if theme_options.get("analytics"):
+            theme_options["analytics"].update({"google_analytics_id": gid})
```

Comment:
This is to account for other providers. You could have Plausible+Google Analytics. I could add a test but straightforward.